### ISSUE TITLE
Zoom Out not going to minimum level of image.

### ISSUE THREAD

**Issue by malik-umair** [openseadragon/openseadragon#1177]
Hi, when i try to zoom out image in viewer it stops after one point and don't allow to zoom out further but when i open same viewer image in browser i can zoom out image as much as i want. As you can see on following page, try to zoom out this image:

    http://baf-website-dev.itcomrade.us/Product/1080

But when you open same image in brower by copy paste following image link then you can zoom out too much.

    http://bafdev.iiifhosting.com/iiif/c6424f9a3657c28f3ca8aa129118f4873887050cfb29b156dab4e24516a9b2fc

Following is my code:

    var osd_config = {
    id: "viewer",
    prefixUrl: prefixUrl,
    preserveViewport: true,
    showNavigator: false,
    visibilityRatio: 1,
    zoomInButton: "zoom-in",
    zoomOutButton: "zoom-out",
    homeButton: "home",
    fullPageButton: "full-page",
    nextButton: "next",
    previousButton: "previous",
    showRotationControl: true,
    gestureSettingsTouch: {
        pinchRotate: true
    },
    tileSources: []
    };

    feedMe = getUrlVars()['feedme'];
    if (feedMe) {
      osd_config['tileSources'].push(image + INFO);
    } else {
      osd_config['tileSources'].push(image + INFO);
    }

    var viewer = OpenSeadragon(osd_config);
    viewer.addHandler('open', function () {
      viewer.viewport.zoomTo(0, null, true);
      viewer.viewport.applyConstraints();
      viewer.viewport.maxZoomLevel = viewer.viewport.getZoom(true) * 8;
    });

**Comment by iangilman** [comment]
You should be able to add:
```
            minZoomImageRatio: 1
```
...to your osd_config and you won't be able to zoom out from the full image.

**Comment by malik-umair** [comment]
@iangilman  Added but nothing changed, still i can't zoom out..

**Comment by malik-umair** [comment]
I noticed when i add minZoomImageRatio:1 it has effected viewer design. I want to have margins from top, bottom left and right when image load as shown in this screenshot http://prntscr.com/f1qaz9 and this space fill up when user zoom in image.

but after adding your mentioned line of code it filled up top, bottom space.

**Comment by iangilman** [comment]
Oh! I'm sorry, I misunderstood... I thought you wanted to not be able to zoom out. By default you can zoom out a little. Also, you can start it off zoomed out a bit like so:

```
            defaultZoomLevel: 0.5
```

Though if you just want some margins above and below the image, it might be best to just put the margins outside the viewer.

**Comment by iangilman** [comment]
Good point! The `defaultZoomLevel` is based on the width of the image, so if you have a tall image it's not necessarily right. Looks like you'll have to calculate it once the image loads like so:

```
        viewer.addHandler('open', function() {
            var targetZoom = 0.9;
            
            var imageBounds = viewer.world.getItemAt(0).getBounds();
            var viewportBounds = viewer.viewport.getBounds();
            var imageAspect = imageBounds.width / imageBounds.height;
            var viewportAspect = viewportBounds.width / viewportBounds.height;
            var aspectFactor = imageAspect / viewportAspect;
            var zoomFactor = (aspectFactor >= 1 ? 1 : aspectFactor) * targetZoom;
            viewer.viewport.defaultZoomLevel = zoomFactor / imageBounds.width;
            viewer.viewport.goHome(true);
        });
```

It seems like maybe we should provide a mechanism for this in OSD, as it's not super easy otherwise (without this code).
